tests/functional/src/main/java/org/mule/functional/junit4/ArtifactFunctionalTestCase.java

```diff
@@ -78,7 +79,12 @@ private static final void setPluginClassLoaders(List<ArtifactClassLoader> artifa
         {
             throw new IllegalStateException("Plugin class loaders were already set, it cannot be set again");
         }
-        pluginClassLoaders = artifactClassLoaders;
+
+        pluginClassLoaders = new ArrayList<>();
+        for (Object artifactClassLoader : artifactClassLoaders)
+        {
+            pluginClassLoaders.add((ClassLoader) artifactClassLoader.getClass().getMethod("getClassLoader").invoke(artifactClassLoader));
```

Comment:
Is not this line assuming that artifactClassLoader is instanceOf ArtifactClassLoader?